service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseSubmissionValidatorServiceTest.java

```diff
@@ -0,0 +1,267 @@
+import uk.gov.hmcts.reform.fpl.enums.OrderType;
+import uk.gov.hmcts.reform.fpl.model.Address;
+import uk.gov.hmcts.reform.fpl.model.Applicant;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.Children;
+import uk.gov.hmcts.reform.fpl.model.GroundsForEPO;
+import uk.gov.hmcts.reform.fpl.model.Hearing;
+import uk.gov.hmcts.reform.fpl.model.Orders;
+import uk.gov.hmcts.reform.fpl.model.common.Document;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(SpringExtension.class)
+class CaseSubmissionValidatorServiceTest {
+
+    private final CaseSubmissionValidatorService service = new CaseSubmissionValidatorService();
+
+    @Test
+    void shouldReturnErrorsIfMandatoryCaseSectionsHaveNotBeenCompleted() {
+        CaseData caseData = CaseData.builder().build();
+        List<String> errors = service.validateCaseDetails(caseData);
+
+        assertThat(errors).contains("- Enter a case name");
+        assertThat(errors).contains("- Select at least one type of order");
+        assertThat(errors).contains("- You need to add details to children");
+        assertThat(errors).contains("- You need to add details to applicant");
+        assertThat(errors).contains("- You need to add details to hearing");
+        assertThat(errors).contains("- Tell us the status of all documents including those that you haven't uploaded");
+    }
+
+    @Test
+    void shouldReturnErrorsIfMandatoryFieldsHaveNotBeenCompleted() {
+        CaseData caseData = CaseData.builder()
+            .caseName("Test case")
+            .children(Children.builder().build())
+            .hearing(Hearing.builder().build())
+            .applicant(Applicant.builder().build())
+            .build();
+
+        List<String> errors = service.validateCaseDetails(caseData);
+
+        assertThat(errors).contains("- Select at least one type of order");
+        assertThat(errors).contains("- Select an option for when you need a hearing");
+        assertThat(errors).contains("- Enter the applicant's full name");
+        assertThat(errors).contains("- Enter the contact's full name");
+        assertThat(errors).contains("- Enter a job title for the contact");
+        assertThat(errors).contains("- Enter a valid address for the contact");
+        assertThat(errors).contains("- Enter at least one telephone number for the contact");
+        assertThat(errors).contains("- Enter an email address for the contact");
+        assertThat(errors).contains("- Tell us the names of all children in the case");
+        assertThat(errors).contains("- Tell us the status of all documents including those that you haven't uploaded");
+    }
+
+    @Test
+    void shouldReturnAnErrorWhenEPOHasBeenSelectedButNoGroundsForTheApplicationProvided() {
+        CaseData caseData = CaseData.builder()
```

Comment:
this seems like a lot of set up for EPO selected but no grounds for application

Reply:
How would you feel about asserting EPO validation message in the validator itself?

Reply:
I would probably suggest having some helper methods giving you case with some details prebuilt, and then method to add all the docs and stuff like that, the code would be less duplicated. Slack me if you are not sure.